### ISSUE TITLE
Validating AKS against external AAD fails to find client service principal

### ISSUE THREAD

**Issue by protomech** [Azure/AKS#1045]
**What happened**:

When validating or deploying an AKS cluster configured to use an external Azure AD, we receive an error stating that the client service principal was not found in the external directory. 

Service principals were created in this external directory according to the instructions here:
https://docs.microsoft.com/en-us/azure/aks/azure-ad-integration-cli

The external directory was initially not linked to a subscription, but the error persists after linking the directory to a subscription (and granting the user validating the AKS cluster template owner on the new subscription).

We tested the same template against service principals created in the same method against a different external directory, and both validation and creating the resources works correctly. So something is different in how AKS resources are created against the two directories.

```
Service principal clientID: a8cf089f-71d0-48e6-b40a-c7ac761d2ed4 not found in Active Directory tenant af5095b9-beba-44ce-8eee-77666ac93d88, Please see https://aka.ms/aks-sp-help for more details.
```

**What you expected to happen**:

An AKS cluster to be created linked to the external directory for authentication / authorization.

**How to reproduce it (as minimally and precisely as possible)**:

The following steps **may** work:
1. Create a subscription.
2. Create a directory, **do not link it to the subscription**.
3. Follow the steps here to create the service principals, resource groups, and cluster:
https://docs.microsoft.com/en-us/azure/aks/azure-ad-integration-cli

**Anything else we need to know?**:

We receive the same error whether deploying an AKS cluster with the CLI or with ARM templates.

Deploying an AKS cluster without AAD integration works fine, as does both deploying an AKS cluster against AAD linked to the deploying subscription as well as an additional external directory.

The failing directory was previously an AD Premium P2 trial, now downgraded to "Azure AD for Office 365".

The working directory is "Azure AD Free".

**Environment**:
- Kubernetes version (use `kubectl version`): 1.11.9
- Size of cluster: 3 Bs2

**Comment by sauryadas** [comment]
can you register container service in the subs of the external directory and check if it works?

az provider register --namespace Microsoft.ContainerService

**Comment by protomech** [comment]
> can you register container service in the subs of the external directory and check if it works?
> 
> az provider register --namespace Microsoft.ContainerService

Oh wow. That worked for us!

We had deployed AKS (non-AAD) on the working external directory, so we must have somehow enabled AKS for the subscription. I gather it's not enabled by default if we had to run this, as we have never deployed AKS in the new subscription.

**Future reference** - run `az provider register --namespace Microsoft.ContainerService` on new clusters!
